components/uniset/src/uniset.rs

```diff
@@ -0,0 +1,366 @@
+use std::{boxed::Box, error::Error, iter::Iterator, slice::Iter, vec::Vec};
+
+/// Represents the maximum Unicode Code Point, inclusive
+const CODEPOINT_MAX: u32 = 0x10FFFF;
+/// Represents the minimum UNicode Code Point, inclusive
+const CODEPOINT_MIN: u32 = 0x000000;
+/// Represents the end code point of the Basic Multilingual Plane range, starting from code point 0 , inclusive
+const BMP_MAX: u32 = 0xFFFF;
+
+/// Given string representation of inversion list create set
+///
+/// See UnicodeSet::new for conditions
+fn parse_serial_string(serialize_str: &str) -> Result<Vec<u32>, Box<dyn Error>> {
+    let mut serialize = serialize_str.split(" ");
```

Comment:
I don't know if this is the right architecture for serializing.  I'm increasingly thinking that the JSON schema should literally be a sorted list of unsigned integers, and in UnicodeSet, you simply construct from that list of integers.  Then you don't need this complicated parsing function, instead deferring the problem to Serde and the data provider.

Reply:
agreed and changed